.github/wiki/Project-Configurations.md

```diff
@@ -0,0 +1,136 @@
+
+- Terminologies and topics related to project configurations.
+- Basic targets, schemes, build settings options, etc., for building a project.
+
+### **Targets**
+
+A target specifies a product to build, such as an iOS, watchOS, or macOS app.
+
+When creating a project from a template, a default target will be added automatically for the main application
+
+There are 3 default targets in a project:
+
+- {ProjectName}
+- {ProjectName}Tests
+- {ProjectName}UITests
+
+### Schemes
+
+A scheme is a collection of settings that specifies the targets to build for a project, the build configuration to use, and the executable environment to use when the product is launched.
+
+The 2 main schemes in a project:
+
+- {ProjectName}
+- {ProjectName} Staging
+
+### Build Configurations in Schemes
+
+It is possible to build a scheme with different Build Configurations. Initially, there are 2 basic configurations for a project generated automatically by Xcode, which are:
+
+- Debug
+- Release
+
+However, these basic configurations are not enough for our development process. Since our team always wants to ensure the application is working in a designated manner, additional levels of testing are required before the application is ready to be used.
+
+The recommended set of configurations to have is:
+
+- Debug Staging
+- Release Staging
+- Debug Production
+- Release Production
+
+### Active Compilation Conditions
+
+Having distinct flags for each configuration is the preferred solution.
+
+Xcode provides a build setting named `Active Compilation Conditions`, which supports passing conditional compilation flags to the Swift compiler.
+
+In our iOS template project, there are three custom conditional compilation flags: `DEBUG`, `STAGING`, `PRODUCTION`. The following table will describe how the developers differentiate a configuration from the others.
+
+| Build Configurations | DEBUG | STAGING | PRODUCTION |
+|---|---|---|---|
+| Debug Staging | ✔︎ | ✔︎ |  |
+| Release Staging |  | ✔︎ |  |
+| Debug Production | ✔︎ |  | ✔︎ |
+| Release Production |  |  | ✔︎ |
+
+The huge advantage when using custom flags is customizable specific features based on a particular environment.
+
+Example: 
+
+Specify a value based on the environment:
+
+```swift
+enum Environment {
+
+    static func based<T>(staging: T, production: T) -> T {
+        #if PRODUCTION
+        return production
+        #else
+        return staging
+        #endif
+	}
+}
+```
+
+During the development process, all functionalities and frameworks are available in the application. However, there can be situations when the developer turns on some features for a particular environment without creating separate modules for the application. In these cases, taking advantage of `Active Compilation Conditions` by declaring a condition to include or exclude code is the recommended solution.
```

Comment:
Small wording improvements. Also, I think we should use `developers` in plural form for consistency with other places' usage in this post 🙏
```suggestion
During the development process, all functionalities and frameworks are available in the application. However, there can be situations when the developers just want to turn on some features for a particular environment without creating separate modules for the application. In these cases, declaring a condition to include or exclude code using `Active Compilation Conditions` is the recommended solution.
```

Reply:
Fixed in https://github.com/nimblehq/ios-templates/pull/268/commits/4ac43d42bd69de88234f85aee75cd8736fdfc56f